src/utils.ts

```diff
@@ -440,57 +451,61 @@ export async function resolveIdxProfile(caip10: string, config: Config): Promise
-      const registration = await getRegistration(name, config, resolver);
-      if (registration) {
-        return registration.profile;
+export const resolveEnsProfile = memoizeLru(
+  async (addressOrName: string, profileType: ProfileType, config: Config): Promise<EnsProfile | null> => {
+    const name = ethers.utils.isAddress(addressOrName)
+      ? await config.provider.lookupAddress(addressOrName)
+      : addressOrName;
+
+    if (name) {
+      const resolver = await config.provider.getResolver(name);
+      if (! resolver) {
+        return null;
       }
-    } else {
-      const promises: [Promise<any>] = [
-        getAvatar(name, config, resolver),
-      ];
 
-      if (addressOrName === name) {
-        promises.push(resolver.getAddress());
+      if (profileType === ProfileType.Full) {
+        const registration = await getRegistration(name, config, resolver);
+        if (registration) {
+          return registration.profile;
+        }
       } else {
-        promises.push(Promise.resolve(addressOrName));
+        const promises: [Promise<any>] = [
+          getAvatar(name, config, resolver),
+        ];
+
+        if (addressOrName === name) {
+          promises.push(resolver.getAddress());
+        } else {
+          promises.push(Promise.resolve(addressOrName));
+        }
+
+        if (profileType === ProfileType.Project) {
+          promises.push(getSeed(name, config, resolver));
+          promises.push(getAnchorsAccount(name, config, resolver));
+        } else if (profileType === ProfileType.Minimal) {
+          promises.push(Promise.resolve(null));
+        }
+
+        const project = await Promise.allSettled(promises);
+        const [avatar, address, seed, anchorsAccount] =
+          // Just checking for r.value equal null and casting to undefined,
+          // since resolver functions return null.
+          project.map(r => r.status == "fulfilled" && r.value ? r.value : null);
+
+        return {
+          name,
+          avatar,
+          address,
+          seed,
+          anchorsAccount,
+        };
       }
-
-      if (profileType === ProfileType.Project) {
-        promises.push(getSeed(name, config, resolver));
-        promises.push(getAnchorsAccount(name, config, resolver));
-      } else if (profileType === ProfileType.Minimal) {
-        promises.push(Promise.resolve(null));
-      }
-
-      const project = await Promise.allSettled(promises);
-      const [avatar, address, seed, anchorsAccount] =
-        // Just checking for r.value equal null and casting to undefined,
-        // since resolver functions return null.
-        project.map(r => r.status == "fulfilled" && r.value ? r.value : null);
-
-      return {
-        name,
-        avatar,
-        address,
-        seed,
-        anchorsAccount,
-      };
     }
-  }
-  return null;
-}
+    return null;
+  },
+  (addressOrName: string, profileType: ProfileType) => JSON.stringify({ addressOrName, profileType }),
```

Comment:
I wonder if we shouldn't also key it by `config.network.chainId`? Do the addresses that we query differ across networks?

Reply:
> I guess if it's guaranteed that there is a page refresh upon network switch then it shouldn't be a problem, cause the cache would be yanked anyway, right?

Yes we are indeed doing a reload with `window.location.reload()` everytime we change the `chainId` here
https://github.com/radicle-dev/radicle-interface/blob/65bf2ae38678caf6e7aa444f9a24bf4efd61f7f8/src/session.ts#L307